src/aws-cpp-sdk-core/source/auth/signer/AWSAuthV4Signer.cpp

```diff
@@ -237,17 +237,18 @@ bool AWSAuthV4Signer::SignRequest(Aws::Http::HttpRequest& request, const char* r
             return false;
         }
 
-        if (request.GetRequestHash().second != nullptr)
+        Aws::String checksumHeaderKey = Aws::String("x-amz-checksum-") + request.GetRequestHash().first;
+        const auto headers = request.GetHeaders();
```

Comment:
accidental copy of all request's headers creation.

Reply:
ah can just use `HasHeader` actually, doing that instead